src/MongoDB.Driver.Core/Core/Misc/DnsClientWrapper.cs

```diff
@@ -38,6 +38,23 @@ public DnsClientWrapper()
         public LookupClient LookupClient => _lookupClient;
 
         // public methods
+        public string GetHostNameWithReverseDnsLookup(string hostNameOrAddress)
+        {
+            var hostEntry = _lookupClient.GetHostEntry(hostNameOrAddress);
+            if (hostEntry != null)
+            {
+                if (hostEntry.Aliases != null && hostEntry.Aliases.Length > 0)
+                {
+                    return hostEntry.Aliases[0];
```

Comment:
Why `Aliases[0]`?

How do we know that's what we need? Why `Aliases[0]` instead of `HostName`?

Reply:
because `HostName` returns `Canonical` host name, but we need to have the canonical host name with reversed DNS resolution. 

Docs:
>GetHostEntry(IDnsQuery, String)
The GetHostEntry method queries a DNS server for the IP addresses and aliases associated with the hostNameOrAddress. In case hostNameOrAddress is an IPAddress, GetHostEntry does a reverse lookup on that first to determine the hostname. 

http://dnsclient.michaco.net/docs/DnsClient.DnsQueryExtensions.html#DnsClient_DnsQueryExtensions_GetHostEntry_DnsClient_IDnsQuery_System_String_

Reply:
So since the values in Aliases are reversed, the first item will contain the highest dns level.

Reply:
What does "reversed" mean?

Are the aliases in some sort of order? Why is `0` the right one?

What does "highest dns level" mean?